website/public/js/controllers/settingsCtrl.js

```diff
@@ -66,12 +66,23 @@ habitrpg.controller('SettingsCtrl',
       User.set({'flags.newStuff':true});
     }
 
-    $scope.saveDayStart = function(){
-      var dayStart = +User.user.preferences.dayStart;
-      if (_.isNaN(dayStart) || dayStart < 0 || dayStart > 24) {
+    $scope.passDayStart = User.user.preferences.dayStart;
+
+    $scope.saveDayStart = function(newDayStart){
+      var oldDayStart = User.user.preferences.dayStart;
+      var dayStart = newDayStart;
+      var lastCron = User.user.lastCron;
+      var getOldStart = Shared.startOfDay({ dayStart: oldDayStart});
+      var getNewStart = Shared.startOfDay({ dayStart: dayStart});
+      var isoNewStart = Shared.isoTimestamp(getNewStart);
+
+      if (dayStart == undefined || _.isNaN(dayStart) || dayStart < 0 || dayStart > 24) {
```

Comment:
Well, this is interesting and brings up more questions. Do we want to allow fractions? Is lodash not working on my machine for some reason? I'm really stumped now. Here is the result of my testing:

testing options for case to kick off message:
"please enter a number between 0 and 24"

if ( <result of test> || dayStart < 0 || dayStart > 24 ) {message}

Test 1: pass in the number zero, desired response = false

1. alert(dayStart); 0
2. alert(!dayStart); true
3. alert(isNaN(dayStart)); false
4. alert(!isNaN(dayStart)); true
5. alert(_.isNaN(dayStart)); false

success: 3,5

Test 2: pass in the number 1, desired response = false

1. alert(dayStart); 1
2. alert(!dayStart); false
3. alert(isNaN(dayStart)); false
4. alert(!isNaN(dayStart)); true
5. alert(_.isNaN(dayStart)); false

success: 2,3,5

Test 3: pass in the number 23, desired response = false

1. alert(dayStart);23
2. alert(!dayStart); false
3. alert(isNaN(dayStart)); false
4. alert(!isNaN(dayStart)); true
5. alert(_.isNaN(dayStart)); false

success: 2,3,5

Test 4: pass in the number -1, desired response = true

1. alert(dayStart); undefined
2. alert(!dayStart); true
3. alert(isNaN(dayStart)); true
4. alert(!isNaN(dayStart)); false
5. alert(_.isNaN(dayStart)); false

success: 2,3


Test 5: pass in the number 5.5, desired response ?

1. alert(dayStart); 5.5
2. alert(!dayStart); false
3. alert(isNaN(dayStart)); false 
4. alert(!isNaN(dayStart)); true
5. alert(_.isNaN(dayStart)); false

???

Test 6: pass in the number 23.5, desired response = ?

1. alert(dayStart); undefined
2. alert(!dayStart); true
3. alert(isNaN(dayStart)); true 
4. alert(!isNaN(dayStart)); false
5. alert(_.isNaN(dayStart)); false

???

Test 7: pass in the number 24, desired response = true

1. alert(dayStart); undefined
2. alert(!dayStart); true
3. alert(isNaN(dayStart)); true
4. alert(!isNaN(dayStart)); false
5. alert(_.isNaN(dayStart)); false

2,3

Test 8: pass in the letter k, desired response = true

1. alert(dayStart); undefined
2. alert(!dayStart); true
3. alert(isNaN(dayStart)); true
4. alert(!isNaN(dayStart)); false
5. alert(_.isNaN(dayStart)); false

2,3

Test 9: remove the value from the page, desired response = true

1. alert(dayStart); null
2. alert(!dayStart); true
3. alert(isNaN(dayStart)); false
4. alert(!isNaN(dayStart)); true
5. alert(_.isNaN(dayStart)); false

success: 2,4